### PR TITLE
Fixed HOSTNAME env variable in Dockerfile

### PR DESCRIPTION
**Pull request by DragosMocrii**
### What?
This change is for the `Example with Docker using Multi Environments` example. Changed the HOSTNAME env variable in Dockerfile for each environment from `localhost` to `0.0.0.0`

### Why?
Using `localhost` as the hostname running the node server will not make the server accessible outside of the container. Using `0.0.0.0` ensure the server is accessible. (Read more: https://serverfault.com/a/1084918)

### CODE REVIEW

**Comment by DuCanhGH** [comment]
The default for `.next/standalone/server.js` is now `0.0.0.0`, so you may as well delete `ENV HOSTNAME` altogether.

**Comment by DragosMocrii** [comment]
> The default for `.next/standalone/server.js` is now `0.0.0.0`, so you may as well delete `ENV HOSTNAME` altogether.

You are right in that the default hostname value for the standalone server is 0.0.0.0, however the current value used in the dockerfile is incorrect, so for one using this example the server will fail to be accessible from the outside of the container.

**Comment by DuCanhGH** [comment]
@DragosMocrii well I mentioned deleting that value, right? Sometimes the deployment platform just provides `HOSTNAME` for you (otherwise the default value `0.0.0.0` should just work), and you'd only need to have `ENV HOSTNAME 0.0.0.0` if that value isn't actually meant to be a hostname. Though, maybe adding a comment like this would be nice:
```
# Uncomment the following line if `process.env.HOSTNAME` is not provided by your provider
# or it is an invalid value for a hostname.
# ENV HOSTNAME 0.0.0.0
```
This is just my suggestion though, feel free to do it how you want :) It'd be nice if you could change this for other Docker examples as well (they should be close to this example).

**Review by leerob — CHANGES_REQUESTED** [review]
Hostname is not needed. It can be removed entirely. Could you update this please?

**Comment by DragosMocrii** [comment]
@DuCanhGH sorry, I missed your initial point. I've made the updates per your suggestion.

@leerob updated

File: examples/with-docker/Dockerfile [hunk 1]
```diff
@@ -59,7 +59,9 @@ USER nextjs
 EXPOSE 3000
 
 ENV PORT 3000
-# set hostname to localhost
-ENV HOSTNAME "0.0.0.0"
+
+# Uncomment the following line if you'd like to change the default value
```
Comment (leerob) [inline examples/with-docker/Dockerfile:63]: Let's just remove it entirely, no commented out block. Thank you!

**Comment by leerob** [comment]
Closing as dupe of https://github.com/vercel/next.js/pull/55239.

**Pull request closed without merging**